S21/cihanozluk/ToLower/ToLower/UpperToLower.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+namespace ToLower
+{
+    public class UpperToLower
+    {
+
+        public UpperToLower(string word, string language)
+        {
+            if (string.IsNullOrEmpty(language))
+            {
+                throw new ArgumentException($"'{nameof(language)}' cannot be null or empty.", nameof(language));
+            }
+            Word = word ?? throw new ArgumentNullException(nameof(word));
+            Language = language;
+
+        }
+
+
+        public string Word { get; }
+
+        public string Language { get; }
+
+
+        private string GetToLower() => Word.ToLower(new System.Globalization.CultureInfo(Language));
+
+        public string ToGetLower()
+        {
+            List<Alphabet> alphabets = new List<Alphabet>();
+            alphabets.Add(new Alphabet("tr", "ABCÇDEFHĞHIİJKLMNOÖPQRSŞTUÜVWXYZ", "abcçdefgğhıijklmnoöpqrsştuüvwxyz"));
+            alphabets.Add(new Alphabet("el", "ΑΒΓΔΕΖΗΘΙΚΛΜΝΞΟΠΡΣΤΥΦΧΨΩ", "αβγδϵζηθικλμνξοπρστυϕχψω"));
+            alphabets.Add(new Alphabet("ga", "ABCDEFGHILMNOPRSTUÁÉÍÓÚ", "abcdefghilmnoprstuáéíóú"));
+            alphabets.Add(new Alphabet("ga-IE", "ABCDEFGHILMNOPRSTUÁÉÍÓÚ", "abcdefghilmnoprstuáéíóú"));
+
+            if (Language != "tr" && Language != "el" && Language != "ga" && Language != "ga-IE")
+                return GetToLower();
+            else
+            {
+                var lowerWord = new StringBuilder();
+                var str = alphabets.Find(r => r.Language == Language);
+
+
+
+                foreach (char item in Word)
+                {
+
+                    lowerWord.Append(GetToLowerChar(item, str));
+                }
+
+                if (Language == "el" && Word.Substring(Word.Length - 1, 1) == "Σ")
+                {
+                    lowerWord[Word.Length - 1] = 'ς';
+                }
+                if ((Language == "ga" || Language == "ga-IE") && (Word.Substring(0, 1) == "n" || (Word.Substring(0, 1) == "t" && str.Upper.IndexOf(Word.Substring(1, 1)) >= 0)))
```

Comment:
1. Comparing the language strings will not satisfy other varieties of 'ga'. For example 'ga-IA', 'ga-IB'(this is just an example)
2. Can you explain last condition with some comment ?

Reply:
last condition checking second char of the word is Uppercase